extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ClassConfigPropertiesUtil.java

```diff
@@ -427,6 +451,10 @@ private static void populateTypicalProperty(MethodCreator methodCreator, ResultH
     }
 
     private static String getFullConfigName(String prefixStr, ConfigProperties.NamingStrategy namingStrategy, FieldInfo field) {
+        return prefixStr + "." + getEffectiveConfigName(namingStrategy, field);
```

Comment:
Will `prefixStr` ever be null or empty?

Reply:
Nope. It's either read from the annotation or deduced from the class name